Added VS RX data path to ISO test

mtpr-ot opened the pull request:
tests: bluetooth: Added VS RX data path to ISO test

Verifies ISOAL functionality for VS RX data path by implementing weak
function ll_data_path_sink_create, and implementing sink callbacks.

Test verifies construction of sink and checks that SDUs are emitted in
VS sink implementation.

Test relies on the Nordic synchronous receiver implementation, and is
compiled under CONFIG_BT_CTLR_ISO_VENDOR_DATA_PATH, which is enabled in
the project file.

Bluetooth: controller: Removed compiler warnings in ull_iso.c

Conditionally declared auto-variables to avoid 'unused' warnings in BSIM
ISO test compilation.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot committed:
tests: bluetooth: Added VS RX data path to ISO test

Verifies ISOAL functionality for VS RX data path by implementing weak
function ll_data_path_sink_create, and implementing sink callbacks.

Test verifies construction of sink and checks that SDUs are emitted in
VS sink implementation.

Test relies on the Nordic synchronous receiver implementation, and is
compiled under CONFIG_BT_CTLR_ISO_VENDOR_DATA_PATH, which is enabled in
the project file.

Signed-off-by: Morten Priess <mtpr@oticon.com>

wopu-ot commented on tests/bluetooth/bsim_bt/bsim_test_iso/src/main.c:
```diff
@@ -609,6 +671,52 @@ static void test_iso_recv_main(void)
 		FAIL("Local Host Terminate Failed.\n");
 	}
 
+#if defined(CONFIG_BT_CTLR_ISO_VENDOR_DATA_PATH)
```
I would have put this sub-test in a separate function, but having everything here seems to follow the general style of this test, so ok.

mtpr-ot replied:
Yeah, good point. Just wanted to piggy-back on BIS setup already in place, but I think I'll move it

mtpr-ot replied:
On second thought, that requires a lot of redundancy in code and execution scripts. I think I'll leave it as is, if @cvinayak approves.

cvinayak replied:
Since I started the mess of having large executing test as a form of test driven development that kept adding the calls to APIs, I do not want to hinder this quick approach to validate a feature being implemented.

There has always been request to clean up the existing large test executions into smaller chunks, this has not been forgotten. I am sure we will always remember and when possibly time permitted improve in small portions.

I have no strong objection here.

cvinayak commented on tests/bluetooth/bsim_bt/bsim_test_iso/prj.conf:
```diff
@@ -18,4 +18,6 @@ CONFIG_BT_CTLR_SYNC_ISO=y
 CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX=251
 CONFIG_BT_CTLR_SYNC_ISO_PDU_LEN_MAX=251
 
+CONFIG_BT_CTLR_ISO_VENDOR_DATA_PATH=y
+
```
Make a new prj_vendor_dp.conf file or some suitable name and let vendor path be a new test run in CI, see the `compile.sh` how LOW LAT variant of controller is tested using existing test application.

mtpr-ot commented:
I opted for restructuring to sepate main and conf file. That also necessitates a new shell script. How do I make CI run my new .sh?

cvinayak commented:
> Is it correct that `run_parallel.sh` will pick it up by extension?

Yes. Your PR has run the new script and has passed: https://github.com/zephyrproject-rtos/zephyr/runs/5316915037?check_suite_focus=true

mtpr-ot commented:
@wopu-ot, can you please take another look. Thanks

mbolivar-nordic commented:
@mtpr-ot please rebase

mtpr-ot committed:
Bluetooth: controller: Removed compiler warnings in ull_iso.c

Conditionally declared auto-variables to avoid 'unused' warnings in BSIM
ISO test compilation.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot commented:
Rebased. @cvinayak and @wopu-ot, please renew your approvals. Thanks.
